### PR TITLE
feat(rpc-types-engine): EIP-1559 parameters in `OptimismPayloadAttributes`

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds the `eip1559Params` field to the `OptimismPayloadAttributes, per https://github.com/ethereum-optimism/specs/pull/404

### CODE REVIEW

File: crates/rpc-types-engine/src/attributes.rs [hunk 1]
```diff
@@ -26,6 +26,11 @@ pub struct OptimismPayloadAttributes {
         serde(skip_serializing_if = "Option::is_none", with = "alloy_serde::quantity::opt")
     )]
     pub gas_limit: Option<u64>,
+    /// If set, this sets the EIP-1559 parameters for the block.
+    ///
+    /// Prior to Holocene activation, this field should always be [None].
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
+    pub eip_1559_params: Option<FixedBytes<8>>,
```
Comment (mattsse) [inline crates/rpc-types-engine/src/attributes.rs:33]: this is u8 and can use alloy_serde::quantity::opt

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/rpc-types-engine/src/attributes.rs (hunk 1 above)
Comment (mattsse) [reply crates/rpc-types-engine/src/attributes.rs:33]: or is it?
maybe not, if this is a hash type then this should be `B8`

Comment (clabby) [reply crates/rpc-types-engine/src/attributes.rs:33]: It's not? It's not - it's a fixed bytes type of `8` bytes in len.

extension is spec'd as:
```
PayloadAttributesV3: {
    timestamp: QUANTITY
    random: DATA (32 bytes)
    suggestedFeeRecipient: DATA (20 bytes)
    withdrawals: array of WithdrawalV1
    parentBeaconBlockRoot: DATA (32 bytes)
    transactions: array of DATA
    noTxPool: bool
    gasLimit: QUANTITY or null
    eip1559Params: DATA (8 bytes)
}
```

we're packing 2x 32 bit values (denominator, elasticity) within it.

Comment (clabby) [reply crates/rpc-types-engine/src/attributes.rs]: changed to `B64`.

File: crates/rpc-types-engine/src/attributes.rs [hunk 2]
```diff
@@ -26,6 +26,11 @@ pub struct OptimismPayloadAttributes {
         serde(skip_serializing_if = "Option::is_none", with = "alloy_serde::quantity::opt")
     )]
     pub gas_limit: Option<u64>,
+    /// If set, this sets the EIP-1559 parameters for the block.
+    ///
+    /// Prior to Holocene activation, this field should always be [None].
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
+    pub eip_1559_params: Option<B64>,
```
Comment (mattsse) [inline crates/rpc-types-engine/src/attributes.rs:33]: could we add a serde roundtrip test where some fields are set to none?

suddenly no longer 100% sure whether we don't actually need serde(default) here

File: crates/rpc-types-engine/src/attributes.rs (hunk 1 above)
Comment (mattsse) [reply crates/rpc-types-engine/src/attributes.rs]: gotcha

File: crates/rpc-types-engine/src/attributes.rs (hunk 2 above)
Comment (refcell) [reply crates/rpc-types-engine/src/attributes.rs:33]: +1

Comment (refcell) [reply crates/rpc-types-engine/src/attributes.rs:33]: Though I don't think we do need a `serde(default)` since this follows the above optional fields.

**Review by refcell — APPROVED** [review]

**Review by mattsse — APPROVED** [review]
